app/models/mixins/process_tasks_mixin.rb

```diff
@@ -116,33 +113,36 @@ def invoke_api_tasks(api_client, remote_options)
 
       collection   = api_client.send(collection_name)
       action       = action_for_task(remote_options[:task])
-      post_args    = remote_options[:args] || {}
       resource_ids = remote_options[:ids]
 
       if resource_ids.present?
         resource_ids.each do |id|
-          begin
-            obj = collection.find(id)
-          rescue ManageIQ::API::Client::ResourceNotFound => err
-            _log.error(err.message)
-          else
-            _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
-            begin
-              obj.send(action, post_args)
-            rescue NoMethodError => err
-              _log.error(err.message)
-            end
-          end
+          send_action(action, collection_name, collection, remote_options, id)
         end
       else
-        _log.info("Invoking task #{action} on collection #{collection_name}, with args #{post_args}")
-        begin
-          collection.send(action, post_args)
-        rescue NoMethodError => err
-          _log.error(err.message)
+        send_action(action, collection_name, collection, remote_options)
+      end
+    end
+
+    def send_action(action, collection_name, destination, remote_options, id = nil)
```

Comment:
I think I would prefer this parameter as `collection` as it is in the caller. Easier to understand I think.

Reply:
👍

Reply:
@carbonin parameter name changed, as well as couple more tests added